tck/src/main/java/org/eclipse/microprofile/reactive/messaging/tck/channel/overflow/BeanUsingLatestOverflowStrategy.java

```diff
@@ -0,0 +1,149 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.eclipse.microprofile.reactive.messaging.tck.channel.overflow;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.reactive.messaging.Channel;
+import org.eclipse.microprofile.reactive.messaging.Emitter;
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import org.eclipse.microprofile.reactive.messaging.OnOverflow;
+import org.eclipse.microprofile.reactive.messaging.Outgoing;
+
+import io.reactivex.Flowable;
+import io.reactivex.Scheduler;
+import io.reactivex.schedulers.Schedulers;
+
+@ApplicationScoped
+public class BeanUsingLatestOverflowStrategy {
+
+    private static ExecutorService executor;
+
+    @PostConstruct
+    public static void init() {
+        executor = Executors.newSingleThreadExecutor();
+    }
+
+    @PreDestroy
+    public static void cleanup() {
+        executor.shutdown();
+    }
+
+    @Inject
+    @Channel("hello")
+    @OnOverflow(value = OnOverflow.Strategy.LATEST)
+    private Emitter<String> emitter;
+
+    private List<String> output = new CopyOnWriteArrayList<>();
+
+    private volatile Throwable downstreamFailure;
+    private volatile boolean done;
+    private Exception callerException;
+
+    public boolean isDone() {
+        return done;
+    }
+
+    public List<String> output() {
+        return output;
+    }
+
+    public Throwable failure() {
+        return downstreamFailure;
+    }
+
+    public Exception exception() {
+        return callerException;
+    }
+
+    public void emitThree() {
+        try {
+            emitter.send("1");
+            emitter.send("2");
+            emitter.send("3");
+            emitter.complete();
+        } 
+        catch (Exception e) {
+            callerException = e;
+        }
+    }
+
+    public void emitALotOfItems() {
+        new Thread(() -> {
+            try {
+                for (int i = 1; i < 1000; i++) {
+                    emitter.send("" + i);
+                }
+            } 
+            catch (Exception e) {
+                callerException = e;
+            } 
+            finally {
+                done = true;
+            }
+        }).start();
+    }
+
+    @Incoming("hello")
+    @Outgoing("out")
+    public Flowable<String> consume(Flowable<String> values) {
+        Scheduler scheduler = Schedulers.from(executor);
```

Comment:
Rx java should not be used.

Reply:
Why can't TCK use rxJava?

Reply:
Because that creates a useless dependency. It should only use Publisher or PublisherBuilder.

Reply:
Also, the Rx java 2 support is really a smallrye implementation details.

Reply:
RxJava2 Flowable is an implementation of Publisher. The TCK code demonstrates a use case though.

Reply:
Understood what @cescoffier as it is not in the spec for supporting the injection of subclass of Publisher etc. I have now updated the tcks.